### PR TITLE
feat:adds details to rest exception

### PR DESCRIPTION
**Pull request by ashish-s**
# Feature #

Add a new property named `details` to `RestException`. `Details` is returned by selective apis to populate extra information about errors. 
The Studio V2 flow api will be returning extra details with flow validation errors. 
### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the master branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: lib/base/RestException.js [hunk 1]
```diff
@@ -4,12 +4,12 @@ class RestException extends Error {
...
     super('[HTTP ' + response.statusCode + '] Failed to execute request');
 
-    const body = response.body === 'string' ? JSON.parse(response.body) : response.body;
+    const body = typeof response.body === 'string' ? JSON.parse(response.body) : response.body;
     this.status = response.statusCode;
     this.message = body.message;
     this.code = body.code;
     this.moreInfo = body.more_info; /* jshint ignore:line */
-    this.detail = body.detail;
+    this.details = body.details === undefined ? null : body.details;
```
Comment (childish-sambino) [inline lib/base/RestException.js:12]: Just `this.details = body.details;` is fine.

Comment (ashish-s) [reply lib/base/RestException.js]: @childish-sambino fixed.

Comment (childish-sambino) [reply lib/base/RestException.js]: Tests are failing now.

Comment (ashish-s) [reply lib/base/RestException.js]: oops sorry, should be fixed now

**Pull request merged**